### PR TITLE
server: be more careful about using not created pods and containers

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

> /kind api-change
/kind bug
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
there exist many chances for us to use a pod or container that hasn't finished creating. 

This PR fixes these issues, and unifies some code paths in finding containers
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Comment by haircommander** [comment]
```
Kubernetes e2e suite: [sig-network] IngressClass [Feature:Ingress] should set default value on new IngressClass [Serial]
```
[hm](https://prow.ci.openshift.org/view/gcs/origin-federated-results/pr-logs/pull/cri-o_cri-o/4238/test_pull_request_crio_e2e_features_rhel/18794/)
/retest

File: server/sandbox_stop_linux.go [hunk 1]
```diff
@@ -24,6 +24,9 @@ func (s *Server) stopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 		if err == sandbox.ErrIDEmpty {
 			return nil, err
 		}
+		if err == errSandboxNotCreated {
```
Comment (mrunalp) [inline server/sandbox_stop_linux.go:27]: Do we have evidence that we get a stop call before we return an id? The more likely scenario is a direct kill of processes  at the cgroup level by the kubelet.

Comment (haircommander) [reply server/sandbox_stop_linux.go:27]: that's a good question. yeah it's not likely the kubelet can get that container....

(I need this PR independent of this race between kill and start btw, but I can put it with the timeout fix if it doesn't make sense alone)

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by mrunalp — APPROVED** [review]

File: server/server.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ const (
 	rootlessEnvName     = "_CRIO_ROOTLESS"
 )
 
+var errSandboxNotCreated = errors.New("sandbox is not created")
```
Comment (kolyshkin) [inline server/server.go:45]: perhaps just
```suggestion
var errSandboxNotCreated = errors.New("sandbox not created")
```

**Comment by haircommander** [comment]
comments addresed!

**Pull request merged**